File: app/src/main/java/com/example/scrolling/di/modules/MyApplicationModule.java
package com.example.scrolling.di.modules;

import com.example.scrolling.ui.list.view.ScrollingActivity;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;

@Module
public abstract class MyApplicationModule {
    @ContributesAndroidInjector
    abstract ScrollingActivity contributeActivityInjector();
}

File: app/src/main/java/com/example/scrolling/di/modules/MyApplicationModule.kt
package com.example.scrolling.di.modules

import android.content.Context
import com.example.scrolling.application.MyApplication
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class MyApplicationModule() {

        @Provides
        @Singleton
        fun providesContext(application: MyApplication): Context {
            return application.applicationContext
        }
}